tutorials/week4-react.md

```diff
@@ -0,0 +1,689 @@
+
+4. Navigate to http://localhost:3000/ to see the default react page.
+    - After a few seconds, a browser window opens, with our app running:
+      ![image](./assets/week4-react/start.png)
+5. [Chakra UI components](https://chakra-ui.com) can be reused and the library can be installed for Next.js with following command:
+    ```bash
+    npm i --save @chakra-ui/react @chakra-ui/next-js
+    ```
+    ![image](./assets/week4-react/chakra.png)
+    > The save flag is used to add the dependency in the package.json file. 
+
+<!-- 
+-   **create-react-app** is a command-line tool that we can use to quickly create a React and TypeScript app with lots of useful pieces.
+
+
+1. We use the create-react-app npm package to create a React and TypeScript project by entering the following:
+
+    ```ts
+    npx create-react-app my-app --template typescript
+    ```
+
+    - Note 1: This will create a new project directory called my-app under the current directory.
+    - Note 2: This will create a git repo in my-app, so you probably shouldn't do this inside a pre-existing git repository.
+
+2. Navigate to the project directory using the command:
+    ```ts
+    cd my-app
+    ```
+3. To start the development server for React, run the command:
+    ```ts
+    npm start
+    ```
+4. Navigate to http://localhost:3000/ to see the default react page.
+    - After a few seconds, a browser window opens, with our app running:
+      ![image](./assets/week4-react/start-screen.png) -->
+
+## React Component
+React follows a Component based architecture. A component is a
+repeatable html element with built-in state, business logic, and a
+lifecycle. The component may be something as simple as a single html
+element such as an input box, or a button, or a complex entity made up
+of other components.
+
+Components are the basic building blocks of a React application and they allow the developer to split the UI into independent and reusable
+pieces where each piece can be used in isolation. 
+### Creating a Function Component in React
+
+<a href="https://codesandbox.io/s/nervous-morse-o3pwqm?file=/src/tutorial/CreateFunctionComponent.tsx" target="_blank">view in sandbox</a>
+
+The simplest method to define a component is to write a function in Javascript. These components are also widely referred as functional stateless components but in the recent versions of React, they have the capabilities to contain state as well.
+```tsx
+import * as React from "react";
+
+interface Props {
+    name?: string
+};
+
+function App (props: Props) {
+  return (
+      <div className="App">
+        <h1>Welcome to React with Typescript Tutorial.</h1>
+      </div>
+  );
+}
+
+export default App;
+```
+
+### A Few Things to Note About React Components:
+
+- The root (App) component is the entry point for the React App and all other components are nested in it.
+- We define a function component using a javascript function, passing the props type in as a generic parameter.
```

Comment:
- We create a functional component by defining a JavaScript function and passing the props type as a generic parameter.